atomic_reactor/plugins/build_source_container.py

```diff
@@ -155,7 +155,7 @@ def run(self):
         shutil.rmtree(image_output_dir)
 
         return BuildResult(
-            logs=output,
```

Comment:
We also return a BuildResult on line 127, shouldn't the output be decoded there as well?

Maybe we could pass `text=True` to the `check_output()` call instead of decoding later?

Reply:
And how come mypy is not failing? BuildResult defines logs as `Optional[List[str]]`

Reply:
> And how come mypy is not failing? BuildResult defines logs as `Optional[List[str]]`

@MartinBasti any idea?

Reply:
Even if we don't figure it out, do we want to do something about it? Fix the type annotation or fix the code to use a list?

Reply:
https://mypy.readthedocs.io/en/stable/getting_started.html#function-signatures-and-dynamic-vs-static-typing
> By default, mypy will not type check dynamically typed functions. This means that with a few exceptions, mypy will not report any errors with regular unannotated Python.

Yeah, so about that "mypy is checking the entire codebase now" - nope :cry:

Reply:
```shell
$ venv/bin/mypy --ignore-missing-imports --check-untyped-defs --package atomic_reactor
...
Found 100 errors in 28 files (checked 83 source files)
```
That's more like it :tada:

Reply:
https://media.giphy.com/media/7p3e2WCM0VEnm/giphy.gif

Reply:
> Even if we don't figure it out, do we want to do something about it? Fix the type annotation or fix the code to use a list?

Yes. It would be good to be in a separate dedicated pr I think. :)

Reply:
> Yeah, so about that "mypy is checking the entire codebase now"...

> `mypy --ignore-missing-imports --check-untyped-defs`

Either that or https://github.com/ben-alkov/actions/pull/11